src/Analytics/Adapter/ActiveCampaign.php

```diff
@@ -40,27 +41,313 @@ class ActiveCampaign extends Adapter
+                'email' => $email
+            ]);
+
+            $result = json_decode($result, true);
+
+            if ($result && $result['meta']['total'] > 0) {
+                return $result['contacts'][0]['id'];
+            } else {
+                return false;
+            }
+        } catch (\Exception $e) {
+            $this->logError($e);
+            return false;
+        }
+    }
+
+    /**
+     * Create a contact
+     * 
+     * @param string $email
+     * @param string $firstName
+     * @param string $lastName
+     * @param string $phone
+     * @return bool
+     */
+    public function createContact(string $email, string $firstName = '', string $lastName = '', string $phone = ''): bool
+    {
+        $body = ['contact' => [
+            'email' => $email,
+            'firstName' => $firstName,
+            'lastName' => $lastName,
+            'phone' => $phone
+        ]];
+
+        try {
+            $this->call('POST', '/api/3/contacts', [
+                'Content-Type' => 'application/json'
+            ], $body);
+            return true;
+        } catch (\Exception $e) {
+            $this->logError($e);
+            return false;
+        }
+    }
+
+    /**
+     * Update contact
+     * 
+     * @param string $contactId
+     * @param string $email
+     * @param string $firstName
+     * @param string $lastName
+     * @param string $phone
+     * 
+     * @return bool
+     */
+    public function updateContact(string $contactId, string $email, string $firstName = '', string $lastName = '', string $phone = ''): bool
+    {
+        $body = ['contact' => [
+            'email' => $email,
+            'firstName' => $firstName,
+            'lastName' => $lastName,
+            'phone' => $phone
+        ]];
+
+        try {
+            $this->call('PUT', '/api/3/contacts/'.$contactId, [
+                'Content-Type' => 'application/json'
+            ], $body);
+
+            return true;
+        } catch (\Exception $e) {
+            $this->logError($e);
+            return false;
+        }
+    }
+
+    /** 
+     * Delete a contact 
+     * 
+     * @param string $email
+     * @return bool
+     */
+    public function deleteContact(string $email): bool {
+        $contact = $this->contactExists($email);
+
+        if (!$contact) {
+            return false;
+        }
+
+        try {
+            $this->call('DELETE', '/api/3/contacts/'.$contact);
+            return true;
+        } catch (\Exception $e) {
+            $this->logError($e);
+            return false;
+        }
+    }
+
+    /**
+     * Account Exists
+     * 
+     * @param string $name
+     * @return bool|int
+     */
+    public function accountExists(string $name): bool|int
+    {
+        try {
+            $result = $this->call('GET', '/api/3/accounts', [], [
+                'search' => $name
+            ]);
+
+            if (intval(json_decode($result, true)['meta']['total']) > 0) {
```

Comment:
Are we sure the ID is always an integer and not a HEX value ?

Reply:
Yes, all ID's for accounts in active campaign are integers in the API Spec